### PR TITLE
Changed ubuntu version

### PR DESCRIPTION
**Pull request by mathewmeconry**
Changed base ubuntu version from LTS 12.04 to LTS 14.04.
This change is need to use libc6 > 2.17 which is needed to use the couchbase driver with nodejs

### CODE REVIEW

**Comment by tleyden** [comment]
Hey @mathewmeconry thanks for the PR!  I'm checking with @mcpaddy from the couchbase server support team to see which versions of couchbase will work with 14.04.

Which version are you running?

**Comment by mathewmeconry** [comment]
Hey @tleyden 
I'm using the community version 4.0 on ubuntu 14.04 for my dev enviroment.
It looks like it works as expected.

**Comment by ceejatec** [comment]
I don't necessarily object to updating the image base, but I want to be sure I understand why. Playing Devil's advocate, at least, it seems like the version of the OS used inside the Couchbase container should be irrelevant to anything outside the container, and nothing other than Couchbase Server should be inside the container.

@mathewmeconry can you explain your use case and what problem you ran into that this sovles?

**Comment by mathewmeconry** [comment]
@ceejatec I know what you mean and understand that I souldn't use anythingelse inside the couchbase container. But to be compatible with our Mac developer who is using a local installation and not docker I installed nodejs inside the container.
But it needs the version 2.16 or higher of libc6 which isn't available for ubuntu 12.04.

**Comment by mcpaddy** [comment]
Couchbase Server 4.X supports Ubuntu 14.04 the older versions do not. That being said docker :whale:  is not an officially support platform . I have no real issue with this change.

>But to be compatible with our Mac developer who is using a local installation and not docker I installed nodejs inside the container.

Would the docker way not be to create a second container with just the nodejs SDK installed? That way you could upgrade the SDK without touching the couchbase container.

**Comment by mathewmeconry** [comment]
@mcpaddy Yes the offical way would be but I don't like to create new config files for the development because every conifg file points in deveolpment to localhost because the main developers doesn't use docker....

**Comment by mcpaddy** [comment]
@tleyden :+1:  from me

**Comment by ceejatec** [comment]
We can't accept this change as-is since the Dockerfile being changed is a generated file. We would need to update the template instead. The same template is used for Couchbase 3.x as well, though, and 3.x didn't support Ubuntu 14.04. So we would need to support a different template for 3.x than 4.x. This starts to sound like a lot of work to support a non-standard use case, and of course other consumers may have different desires for what base image we should be using.

tl;dr: IMHO we should not make this change. @mathewmeconry should be able to "docker build" his local Dockerfile to create an image that solves his problem.

**Comment by mathewmeconry** [comment]
@ceejatec I didn't know about the template. So maybe I can help to change the template for couchbase 4.x and newer versions or we leave it and maybe update it in year, when there is no support for 12.02 anymore.

**Comment by tleyden** [comment]
@ceejatec @mathewmeconry I can change the template, and only have it use ubuntu 14.04 for couchbase server 4.x

**Comment by tleyden** [comment]
>  I can change the template, and only have it use ubuntu 14.04 for couchbase server 4.x

commit https://github.com/couchbase/docker/commit/30cfda6e59fa875afdead26c1a848301a4ef1faf

I will kick off a rebuild of the 4.x docker images.

**Comment by ceejatec** [comment]
Fair enough if it works - thanks Traun! My concern was that we could hit situations in future that we need to make different changes to the template for ubuntu 12.04 vs. 14.04, and end up needing a separate templates. But hopefully that's just me being a worry-wart.

**Comment by tleyden** [comment]
The template is parameterized, and there's some basic logic to decide which ubuntu version to base off of:

```
	case ProductServer:
		if strings.HasPrefix(variant.Version, "2") || strings.HasPrefix(variant.Version, "3") {
			return "ubuntu:12.04"
		}
		return "ubuntu:14.04"
```

**Comment by mathewmeconry** [comment]
thanks for your support guys :+1:

**Comment by tleyden** [comment]
The dockerhub build is queued up: https://hub.docker.com/r/couchbase/server/builds/bnxk6hpyqwdp2dhpv2kkaaa/

**Comment by mathewmeconry** [comment]
@tleyden @ceejatec thanks for your support
